themes/bootstrap3/templates/search/bulk-action-buttons.phtml

```diff
@@ -6,17 +9,82 @@
+        <?=
+          $this->bulkAction()->button(
+              'email',
+              'send-email',
+              'Email',
+              array_merge(
+                  [
+                    'id' => $this->idPrefix . 'ribbon-email',
+                    'title' => $this->translate('bookbag_email_selected'),
+                  ],
+                  $this->formAttr ? ['form' => $this->formAttr] : []
+              )
+          )
+        ?>
         <?php $exportOptions = $this->export()->getActiveFormats('bulk'); ?>
         <?php if (count($exportOptions) > 0): ?>
-          <li><button id="<?=$this->idPrefix?>ribbon-export" class="toolbar-btn" type="submit" name="export" value="1" title="<?=$this->transEscAttr('bookbag_export_selected')?>"<?php if ($this->formAttr): ?> form="<?=$this->escapeHtmlAttr($this->formAttr) ?>"<?php endif; ?>><?=$this->icon('export') ?> <?=$this->transEsc('Export')?></button></li>
+          <?=
+            $this->bulkAction()->button(
+                'export',
+                'export',
+                'Export',
+                array_merge(
+                    [
+                      'id' => $this->idPrefix . 'ribbon-export',
+                      'title' => $this->translate('bookbag_export_selected'),
+                    ],
+                    $this->formAttr ? ['form' => $this->formAttr] : []
+                )
+            )
+          ?>
         <?php endif; ?>
-        <li><button id="<?=$this->idPrefix?>ribbon-print" class="toolbar-btn" type="submit" name="print" value="1" title="<?=$this->transEscAttr('bookbag_print_selected')?>"<?php if ($this->formAttr): ?> form="<?=$this->escapeHtmlAttr($this->formAttr) ?>"<?php endif; ?>><?=$this->icon('print') ?> <?=$this->transEsc('Print')?></button></li>
-        <?php if ($this->userlist()->getMode() !== 'disabled'): ?>
-          <li><button id="<?=$this->idPrefix?>ribbon-save" class="toolbar-btn" type="submit" name="saveCart" value="1" title="<?=$this->transEscAttr('bookbag_save_selected')?>"<?php if ($this->formAttr): ?> form="<?=$this->escapeHtmlAttr($this->formAttr) ?>"<?php endif; ?>><?=$this->icon('user-list-add') ?> <?=$this->transEsc('Save')?></button></li>
+        <?=
+          $this->bulkAction()->button(
+              'print',
+              'print',
+              'Print',
+              array_merge(
+                  [
+                    'id' => $this->idPrefix . 'ribbon-print',
+                    'title' => $this->translate('bookbag_print_selected'),
+                  ],
+                  $this->formAttr ? ['form' => $this->formAttr] : []
+              )
+          )
+        ?>
+        <?php if (($actionLimits?->saveCart ?? 0) > 0 && $this->userlist()->getMode() !== 'disabled'): ?>
```

Comment:
$actionLimits is still being defined without being used in several templates... and it's used here, but I don't think it needs to be because the view helper now incorporates the limit check.

Reply:
cleaned up the templates.